konnectivity-client/go.mod

```diff
@@ -3,8 +3,8 @@ module sigs.k8s.io/apiserver-network-proxy/konnectivity-client
 go 1.17
 
 require (
-	github.com/golang/protobuf v1.4.3
-	google.golang.org/grpc v1.42.0
+	github.com/golang/protobuf v1.4.1
+	google.golang.org/grpc v1.27.0
```

Comment:
https://github.com/kubernetes/kubernetes/blob/master/go.mod#L102 shows v1.40.0 . Would it make more sense to make that our version?

Reply:
(Ditto https://github.com/kubernetes/kubernetes/blob/release-1.23/go.mod#L102)

Reply:
Hmm, they pasted the grpc versions used in different kube versions we want to backport and from their message I understand that we need the lowest here so we can backport without issues.

Am I missing something?

Reply:
There is tension between wanting newer konnectivity-client grpc (to reduce version skew between KAS and proxy-server), and keeping older konnectivity-client grpc (keeping the client backport friendly).

As a balance/compromise, updated to support backport to 1.21.